### PR TITLE
avro-c: init at 1.8.2

### PR DESCRIPTION
**Pull request by lblasc**
###### Motivation for this change

Add C avro bindings.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by lblasc** [comment]
@GrahamcOfBorg build avro-c

**Comment by Mic92** [comment]
@GrahamcOfBorg build avro-c

**Comment by Mic92** [comment]
I suppose you have to patch shebangs before in `postPatch` phase using `patchShebangs .`
https://github.com/apache/avro/blob/master/lang/c/version.sh#L1

File: pkgs/development/libraries/avro-c/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  #
+  src = fetchurl {
+    url = "mirror://apache/avro/avro-${version}/c/avro-c-${version}.tar.gz";
+    sha256 = "03pixl345kkpn1jds03rpdcwjabi41rgdzi8f7y93gcg5cmrhfa6";
+  };
+
+  patchPhase = ''
+    substituteInPlace version.sh \
+      --replace /bin/bash "$bash/bin/bash"
```
Comment (Mic92) [inline pkgs/development/libraries/avro-c/default.nix]: ah weird, you already did this.

**Comment by Mic92** [comment]
@GrahamcOfBorg build avro-c

File: pkgs/development/libraries/avro-c/default.nix [hunk 2]
```diff
@@ -1,35 +1,30 @@
...
   src = fetchurl {
     url = "mirror://apache/avro/avro-${version}/c/avro-c-${version}.tar.gz";
     sha256 = "03pixl345kkpn1jds03rpdcwjabi41rgdzi8f7y93gcg5cmrhfa6";
   };
 
-  patchPhase = ''
-    substituteInPlace version.sh \
-      --replace /bin/bash "$bash/bin/bash"
+  postPatch = ''
+    patchShebangs .
```
Comment (Mic92) [inline pkgs/development/libraries/avro-c/default.nix]: This fixed the build for me.

**Pull request merged**